File: lessons/lesson1/perm_missing_elem.py
def solution(A):
	N = len(A)
	B = [0] * (N + 2)
	
	for i in range(N):
		B[A[i]] = 1
	
	for j in range(1, N+2):
		if B[j] == 0:
			return j

File: java/src/lessons/three/complexity/PermMissingElem.java
package lessons.three.complexity;

public class PermMissingElem {

	public PermMissingElem() {
	}

	public int solution(int[] A) {
		int N = A.length + 1;
		double sumN = (1.0 + N) * N / 2.0 ;
		
		for (int i = 0; i < A.length; i++) {
			sumN -= A[i];
		}
		
		return (int) sumN;
	}
	
}